### ISSUE TITLE
Separate the auto suggest of the current class from the ancestral classes

### ISSUE THREAD

**Issue by gavr123456789** [nim-lang/nimsuggest#121]
Here what I mean:  
```nim
type
  Foo = object of RootObj
  Bar = object of Foo
  Sas = object of Bar


proc foo1(f: Foo, y: int, z: int) =
  echo y, z

proc foo2(f: Foo, y: int, z: int) =
  echo y, z

proc foo3(f: Foo, y: int, z: int) =
  echo y, z

proc bar1(f: Bar, y: int, z: int) =
  echo y, z

proc bar2(f: Bar, y: int, z: int) =
  echo y, z

proc bar3(f: Bar, y: int, z: int) =
  echo y, z

proc sas1(f: Sas, y: int, z: int) =
  echo y, z

proc sas2(f: Sas, y: int, z: int) =
  echo y, z

proc sas3(f: Sas, y: int, z: int) =
  echo y, z


let sas = Sas()

sas.
```  
Here I see all procs in suggestion:  
![image](https://user-images.githubusercontent.com/30507409/114318456-b5b3a580-9b15-11eb-9dab-c699fd052875.png)
  
In large libraries such as GTK, because of the UFCS hints and methods of all the ancestors, a combinatorial explosion of autocompletion is obtained. After each dot, I get most of the GTK as a result of which the auto-completion becomes less useful.  

![image](https://user-images.githubusercontent.com/30507409/114318758-2c04d780-9b17-11eb-8298-c15f3001a890.png)  

I suggest putting the options that relate directly to the object first and marking them in a special way.   
For the example above, this would mean seeing `sas 1,2,3` first, you can also sort by inheritance order, then the order would be `sasN`, `barN`, `fooN`.     
  
  Most likely this issue also applies to https://github.com/pragmagic/vscode-nim

**Comment by Araq** [comment]
How about this alternative: If there are multiple options *of the same name*, only show one. The one that is closest to `sas`'s class.

**Comment by gavr123456789** [comment]
@Araq Can you give an example of what you mean? The _same names_ are functions with an overload? In my example, the functions are named so only to make it clearer which classes they belong to. The main problem is to determine which functions from the suggested ones are directly related to the object after which the point was put.  
  
In the example with GTK, more than 100 functions appear in autocompletion for most objects due to the fact that they are mixed into one heap: methods of objects, methods of the object's ancestors, and simply methods that take any of these entities as the first argument.

**Comment by gavr123456789** [comment]
I will clarify a little, with the inheritance hierarchy A - > B - > C. After the dot after `c`, the first will be members (procedures and fields) of class C, then B and A.

**Comment by Araq** [comment]
>  The same names are functions with an overload?

That's what I mean, yes.

**Comment by gavr123456789** [comment]
> > The same names are functions with an overload?
> 
> That's what I mean, yes.

@Araq  Yes, this is a good idea, in vsc you can display overloads without displaying the same names in the autocomplete  
![image](https://user-images.githubusercontent.com/30507409/114774363-ca43a800-9d78-11eb-8f7c-0eb79f57b382.png)
![image](https://user-images.githubusercontent.com/30507409/114774459-e34c5900-9d78-11eb-951b-b88495536f11.png)
  
But this does not solve the problem of the huge number of procs in auto-completion due to the UFSC. you need to put the procs belonging to the objects in the order of the hierarchy to simplify navigation.  
  
  
![image](https://user-images.githubusercontent.com/30507409/114777413-55726d00-9d7c-11eb-9bcb-982fcf44870b.png)
